pkg/api/controllers/fileops.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2019 OpenSDS Authors All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound service to upload and download configuration files
+
+*/
+
+package controllers
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+
+	log "github.com/golang/glog"
+
+	"github.com/opensds/opensds/pkg/controller/client"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// prometheus constants
+var PROMETHEUS_CONF_HOME string
+var PROMETHEUS_URL string
+var PROMETHEUS_CONF_FILE string
+
+// alert manager constants
+var ALERTMGR_CONF_HOME string
+var ALERTMGR_URL string
+var ALERTMGR_CONF_FILE string
+
+var GRAFANA_CONF_HOME string
+var GRAFANA_RESTART_CMD string
+var GRAFANA_CONF_FILE string
+
+var RELOAD_PATH string
+var BACKUP_EXTENSION string
+
+func init() {
+
+	// TODO Prakash read these from conf and save to these variables
+	RELOAD_PATH = "/-/reload"
+	BACKUP_EXTENSION = ".bak"
+
+	PROMETHEUS_CONF_HOME = "/etc/prometheus/"
+	PROMETHEUS_URL = "http://localhost:9090"
+	PROMETHEUS_CONF_FILE = "prometheus.yml"
+
+	ALERTMGR_CONF_HOME = "/root/alertmanager-0.16.2.linux-amd64/"
+	ALERTMGR_URL = "http://localhost:9093"
+	ALERTMGR_CONF_FILE = "alertmanager.yml"
+
+	GRAFANA_CONF_HOME = "/etc/grafana/"
+	GRAFANA_RESTART_CMD = "grafana-server"
+	GRAFANA_CONF_FILE = "grafana.ini"
+}
+
+func NewFileOpsPortal() *FileOpsPortal {
+	return &FileOpsPortal{
+		CtrClient: client.NewClient(),
+	}
+}
+
+type FileOpsPortal struct {
+	BasePortal
+
+	CtrClient client.Client
+}
+
+func (this *FileOpsPortal) UploadConfFile() {
```

Comment:
Please don't use `this` as the pointer value, change it to `f`.

Reply:
changed